description.md

```diff
@@ -17,6 +17,157 @@ Each section details how a specific type of content is to be developed, stored a
+The contents of the category consist of the text beneath the header.
+For multiple-choice questions, correct answers are marked as list entries starting with `+`, while incorrect ones are marked with `-`.
+
+Every question will be stored in a separate Markdown file.
+See the example below:
+
+```text
+# Sample Question Name
+
+##  Difficulty
+
+easy
+
+## Tags
+
+- topic-1
+
+- topic-2
+
+## Question Text
+
+Is this a sample question?
+
+## Question Answers
+
++ Yes
+
+- No
+
+- Maybe
+
+## Feedback
+
+The question is self-reflecting and its title is "Sample Question Name", so it is a sample question.
+```
+
+Note that correct answers are marked as list entries starting with `+`, while incorrect ones are marked with `-`.
+There may be multiple correct answers.
+In addition, the headings starting with `##` are just keys for different sections of the quiz.
+Therefore, none of them are mandatory given our format and the developer is free to add or remove any of them.
+
+#### Converter
+
+Given the format above, we are providing a converter that can reformat quizzes from one format to another.
+It is bidirectional, so it can translate from Markdown to [Moodle XML](https://docs.moodle.org/402/en/Moodle_XML_format), for example, and vice-versa.
+In addition, it is also capable of translating between non-Markdown formats, such as [Moodle XML](https://docs.moodle.org/402/en/Moodle_XML_format) and SmileyTutor.
+If a new quiz format is desired, the converter may be easily extendable to accommodate it and convert between it and any other format.
+
+The converter achieves this level of flexibility by internally representing each quiz as a dictionary, from which any other format can be derived.
+The keys of this dictionary are the `##` headings and the value of each key is the text beneath the heading.
+This also allows the converter to support a [JSON format](https://blog.hubspot.com/website/json-files) for free, as it is directly mapped to the converter's internal representation.
+For example, the sample question above will be represented as a JSON as follows:
+
+```json
+"Sample Question Name": {
+    "Difficulty": "easy",
+    "Tags": ["topic-1", "topic-2"],
+    "Question Text": "Is this a sample question?",
+    "Question Answers": ["Yes", "No", "Maybe"],
+    "Correct Answer": "Yes",
+    "Feedback": "The question is self-reflecting and its title is \"Sample Question Name\", so it is a sample question."
+}
+```
+
+To use this converter, a person is required to run the command below.
+The script will infer the fact that the input format is Markdown and the output is Moodle XML from the extensions of the input files (`.md` and `.mxml`).
+
+```console
+python3 quiz_converter.py convert -i input_file.md -o output_file.mxml
+```
+
+The converter may also be run by the builder if the teacher desires to integrate quizzes within their published course and not use a third party hosting website, such as those outlined [above](#deployment).
```

Comment:
This is vague. Is it run automatically or it is up to the teacher to configure it in the builder?

Reply:
Not sure whether the builder should provide this functionality off-the-shelf or be left up to the teacher.